src/interactive/Dropdown.jsx

```diff
@@ -1,68 +1,110 @@
-import PropTypes from 'prop-types';
+// @flow
 import React from 'react';
 import cx from 'classnames';
 import Downshift from 'downshift';
 import FloatingPosition from '../utils/components/FloatingPosition';
 import { C_COOLGRAYLIGHTTRANSP } from 'swarm-constants/dist/js/constants.js';
 
-import bindAll from '../utils/bindAll';
-
 export const DROPDOWN_MENU_ITEM_CLASS = 'dropdownMenu-item';
 
-/**
- * @module Dropdown
- */
-class Dropdown extends React.PureComponent {
-	constructor(props) {
-		super(props);
-
-		bindAll(
-			this,
-			'toggleContent',
-			'closeContent',
-			'onClick',
-			'onKeyDown',
-			'onBodyClick',
-			'onBodyKeyDown'
-		);
+type Props = {
+	/** The element that opens the dropdown when clicked */
+	trigger: React$Node,
 
-		this.state = {
-			isActive: props.isActive || false,
-		};
-	}
+	/** The content that's rendered inside the dropdown */
+	content?: React$Node,
+
+	/** An array of elements that are rendered as a menu inside the dropdown */
+	menuItems?: Array<React$Element<*>>,
+
+	/** The horizontal alignment of the dropdown content bubble to the dropdown trigger */
+	align: 'left' | 'right' | 'center',
+
+	/** Which side of the dropdown trigger the dropdown content bubble renders on */
+	direction: 'top' | 'bottom',
+
+	/** How many additional pixels to push the dropdown content bubble */
+	offset?: {
+		left: number,
+		top: number,
+	},
+
+	/** Class names to add to the wrapper of the dropdown trigger and content */
+	className?: string,
+
+	/** Whether the dropdown content is being shown */
+	isActive?: boolean,
+
+	/** A function that is used to control the toggling of the dropdown */
+	manualToggle?: (e: SyntheticKeyboardEvent<*> | SyntheticMouseEvent<*>) => void,
 
-	closeContent(e) {
+	/** The largest width the dropdown content can be */
+	maxWidth?: string | number,
+
+	/** The smallest width the dropdown content can be */
+	minWidth?: string | number,
+
+	/** Whether to render the dropdown content directly in the component instead of bulling it out and attaching to the document root */
```

Comment:
typo 'bulling'. might have meant 'building', but that's sort of an ambiguous way to describe rendering